File: Core/V1/Module/Github/Model/Variable.php
<?php

namespace Core\Github\Model;

class Variable
{
    private string $name;
    private string $value;

    /**
     * Variable constructor.
     *
     * @param string $name
     * @param string $value
     */
    public function __construct(string $name, string $value)
    {
        $this->name = $name;
        $this->value = $value;
    }

    /**
     * Get the name.
     *
     * @return string
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * Get the value.
     *
     * @return string
     */
    public function getValue(): string
    {
        return $this->value;
    }

    /**
     * Serialize the Variable object to JSON.
     *
     * @return string
     */
    public function serialize(): string
    {
        return json_encode([
            'name' => $this->name,
            'value' => $this->value,
        ]);
    }
}

File: Core/Lib/nanorest/src/main/java/org/sindria/xpipe/core/lib/nanorest/github/model/Variable.java
package org.sindria.xpipe.core.lib.nanorest.github.model;

import org.json.JSONObject;
import org.sindria.xpipe.core.lib.nanorest.serializer.JsonSerializer;

public class Variable {

    private final String name;

    private final String value;


    public Variable(String name, String value) {
        this.name = name;
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public String getValue() {
        return value;
    }

    public JSONObject serialize() {
        return new JSONObject(JsonSerializer.toJson(this));
    }

}
